### PR TITLE
Allow domain to contain http scheme

### PR DESCRIPTION
**Pull request by danmastrowcoles**
### Changes
- Add domain scheme as an optional parameter to Authentication.

### References

Resolves https://github.com/auth0/auth0.js/issues/1134

### Testing

- [X] This change adds unit test coverage
- [ ] This change adds integration test coverage

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All tests and linters described in the [Develop section](https://github.com/auth0/auth0.js#develop) run without errors

### CODE REVIEW

**Comment by cocojoe** [comment]
@stevehobbsdev can you take a quick look at this please and validate, stalebot seems quite aggressive here.

File: src/authentication/index.js [hunk 1]
```diff
@@ -88,8 +93,9 @@ function Authentication(auth0, options) {
     this.baseOptions._sendTelemetry === false
       ? this.baseOptions._sendTelemetry
       : true;
-
-  this.baseOptions.rootUrl = 'https://' + this.baseOptions.domain;
+  
+  this.baseOptions.domainScheme = !this.baseOptions.domainScheme ? 'https://' : this.baseOptions.domainScheme
+  this.baseOptions.rootUrl = this.baseOptions.domainScheme + this.baseOptions.domain;
```
Comment (frederikprijck) [inline src/authentication/index.js:98]: What if we would just allow for the domain to include http(s)?

Something along the lines of:

```
this.baseOptions.rootUrl = this.baseOptions.domain.indexOf('http') === 0
  ? this.baseOptions.domain
  : 'https://' + this.baseOptions.domain;
```

This avoids the need for the new option, `domainScheme`, while keeping backwards compatibility.

**Review by frederikprijck — CHANGES_REQUESTED** [review]

File: src/authentication/index.js (hunk 1 above)
Comment (stevehobbsdev) [reply src/authentication/index.js:98]: Agreed, I think auto appending a default scheme if either `http` or `https` was not found would be preferable to having an additional option, unless @danmastrowcoles there is a use case we're not considering here.

Comment (danmastrowcoles) [reply src/authentication/index.js]: Great suggestions, I've refactored to no longer use the additonal option now

File: src/authentication/index.js [hunk 2]
```diff
@@ -88,8 +88,12 @@ function Authentication(auth0, options) {
     this.baseOptions._sendTelemetry === false
       ? this.baseOptions._sendTelemetry
       : true;
+  
 
-  this.baseOptions.rootUrl = 'https://' + this.baseOptions.domain;
+    
+  this.baseOptions.rootUrl = (this.baseOptions.domain && this.baseOptions.domain.toLowerCase().startsWith('http'))
```
Comment (frederikprijck) [inline src/authentication/index.js:94]: We should not use `startsWith`, that is not supported in IE. Please use `indexOf === 0`.

**Review by frederikprijck — CHANGES_REQUESTED** [review]

File: src/authentication/index.js (hunk 2 above)
Comment (danmastrowcoles) [reply src/authentication/index.js:94]: Wow didn't know that, thanks, fixed!

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**